### ISSUE TITLE
rstToHtml: Option to not throw error when parsing

### ISSUE THREAD

**Issue by jyapayne** [nim-lang/Nim#8615]
Discussed in this issue from the nimforum: https://github.com/nim-lang/nimforum/issues/195

Basically, it would be awesome if rstToHtml could handle erroneous data without throwing an exception so that if the user makes an error in the RST, we can still show the generated RST up to a point in the code preview on the forum.

For example, if a stray backtick is entered, instead of erroring out, maybe the rstToHtml function could take in an `RstParseOption` like `NoErrors` or some other name that allows errors to happen but can still render. Maybe an easy solution would be to simply add a backtick at the end of the RST if one doesn't exist.

**Comment by dom96** [comment]
Yeah, this is the way to go. This will also allow us to highlight search results without giving up completely (https://github.com/nim-lang/nimforum/issues/150)
